src/typing/macroContext.ml

```diff
@@ -491,7 +491,13 @@ let get_macro_context ctx p =
 		let mctx = ctx.g.do_create com2 in
 		mctx.is_display_file <- false;
 		create_macro_interp ctx mctx;
-		Option.may (fun cs -> CompilationServer.maybe_add_context_sign cs com2 "get_macro_context") (CompilationServer.get());
+		begin match CompilationServer.get() with
```

Comment:
This should be moved to a separate function as it repeats the code from `typeload.ml` change

Reply:
Good point, done in https://github.com/HaxeFoundation/haxe/commit/1df1daaca7f45949d27fedbe9334149548262ad0